orchagent/neighorch.cpp

```diff
@@ -252,6 +263,15 @@ bool NeighOrch::addNeighbor(NeighborEntry neighborEntry, MacAddress macAddress)
         SWSS_LOG_NOTICE("Created neighbor %s on %s", macAddress.to_string().c_str(), alias.c_str());
         m_intfsOrch->increaseRouterIntfsRefCount(alias);
 
+        if (neighbor_entry.ip_address.addr_family == SAI_IP_ADDR_FAMILY_IPV4)
```

Comment:
This can be *after* the below condition `if (!addNextHop(ip_address, alias)) ` so that if the remove_neighbor_entry is invoked, we don't have to update the resource counter. You can avoid the "decCrmResUsedCounter" inside the "if" below.